onnxruntime.sh

```diff
@@ -19,24 +19,70 @@ prepend_path:
+    export ORT_MIGRAPHX_BUILD=1
+else
+    export ORT_MIGRAPHX_BUILD=0
+fi
+### TensorRT
+if [ "$ORT_CUDA_BUILD" -eq 1 ] && [ $(find /opt/rocm* -name "libnvinfer*" -print -quit | wc -l 2>&1) -eq 1 ]; then
+    export ORT_TENSORRT_BUILD=1
+else
+    export ORT_TENSORRT_BUILD=0
+fi
+
+cmake "$SOURCEDIR/cmake"                                                                                    \
+      -DCMAKE_INSTALL_PREFIX=$INSTALLROOT                                                                   \
+      -DCMAKE_BUILD_TYPE=Release                                                                            \
+      -DCMAKE_INSTALL_LIBDIR=lib                                                                            \
+      -DPYTHON_EXECUTABLE=$(python3 -c "import sys; print(sys.executable)")                                 \
+      -Donnxruntime_BUILD_UNIT_TESTS=OFF                                                                    \
+      -Donnxruntime_PREFER_SYSTEM_LIB=ON                                                                    \
+      -Donnxruntime_BUILD_SHARED_LIB=ON                                                                     \
+      -DProtobuf_USE_STATIC_LIBS=ON                                                                         \
+      -Donnxruntime_ENABLE_TRAINING=OFF                                                                     \
+      ${PROTOBUF_ROOT:+-DProtobuf_LIBRARY=$PROTOBUF_ROOT/lib/libprotobuf.a}                                 \
+      ${PROTOBUF_ROOT:+-DProtobuf_LITE_LIBRARY=$PROTOBUF_ROOT/lib/libprotobuf-lite.a}                       \
+      ${PROTOBUF_ROOT:+-DProtobuf_PROTOC_LIBRARY=$PROTOBUF_ROOT/lib/libprotoc.a}                            \
+      ${PROTOBUF_ROOT:+-DProtobuf_INCLUDE_DIR=$PROTOBUF_ROOT/include}                                       \
+      ${PROTOBUF_ROOT:+-DProtobuf_PROTOC_EXECUTABLE=$PROTOBUF_ROOT/bin/protoc}                              \
+      ${RE2_ROOT:+-DRE2_INCLUDE_DIR=${RE2_ROOT}/include}                                                    \
+      ${BOOST_ROOT:+-DBOOST_INCLUDE_DIR=${BOOST_ROOT}/include}                                              \
+      -Donnxruntime_USE_MIGRAPHX=${ORT_MIGRAPHX_BUILD}                                                      \
+      -Donnxruntime_USE_ROCM=${ORT_ROCM_BUILD}                                                              \
+      -Donnxruntime_ROCM_HOME=/opt/rocm                                                                     \
+      -Donnxruntime_CUDA_HOME=/usr/local/cuda                                                               \
+      -DCMAKE_HIP_COMPILER=/opt/rocm/llvm/bin/clang++                                                       \
+      -D__HIP_PLATFORM_AMD__=1                                                                              \
+      -DCMAKE_HIP_ARCHITECTURES=gfx906,gfx908                                                               \
```

Comment:
I don't understand this. You first set CMAKE_HIP_ARCHITECTURES, and then you possibly override it in the next line? Why don't you expand ALIBUILD_O2_OVERRIDE_HIP_ARCHS to the defaults if empty? `${...:-default}`?